riscv/sim.cc

```diff
@@ -28,9 +28,8 @@ static void handle_signal(int sig)
   signal(sig, &handle_signal);
 }
 
-sim_t::sim_t(const char* isa_string, const char* priv, const char* varch,
-             size_t nprocs, bool halted, bool real_time_clint,
-             reg_t initrd_start, reg_t initrd_end, const char* bootargs,
+sim_t::sim_t(const base_cfg_t *base_cfg,
```

Comment:
Well, my logic was that we're going to end up configuring Spike from some command line parameters plus, possibly, the contents of a DTB file. The command line parameters don't let you specify everything you might want to specify, so I was trying to think of a good name for the "thing that holds the basic configuration".

I'm not wedded to the name though! What do you think might be a better option?

Reply:
How about a `config_t` containing all of the required configuration params; it can be constructed from (1) a DTB, (2) command-line args, or (3) by merging two other `config_t` instances? In the third constructor, you can raise errors if the two provided configs have conflicting non-default values.

Reply:
That's a good idea: let's do that. I think the "incremental" change for this PR is probably just renaming `base_cfg_t` to `cfg_t`, right? If so, I'll do that when we've figured out how the getters will work above.